src/CabanaPD_Solver.hpp

```diff
@@ -111,13 +111,36 @@ class SolverElastic
 
     // Optional module types.
     using heat_transfer_type = HeatTransfer<memory_space, force_model_type>;
+    using contact_type = Force<memory_space, ContactModel>;
+    using contact_model_type = ContactModel;
 
     SolverElastic( input_type _inputs,
                    std::shared_ptr<particle_type> _particles,
                    force_model_type force_model )
         : inputs( _inputs )
         , particles( _particles )
         , _init_time( 0.0 )
+    {
+        setup( force_model );
+    }
+
+    SolverElastic( input_type _inputs,
```

Comment:
Suppose this constructor is explicitly called with `NoContact` tag. I think I am still unsure how best to separate the three possibilities: force_model only, force_model+contact_model, and contact_model only.

Reply:
I agree on not having a good way to support all three clearly. But passing a tag anywhere in the code (while it may seem intuitive) will break everywhere